docs_src/design/adr/0026-Common Configuration.md

```diff
@@ -0,0 +1,393 @@
+# Common Configuration
+### Submitters
+- Lenny Goodell (Intel)
+- Elizabeth Lee (Intel)
+
+## Change Log
+- [pending](URL of PR TBD) (2022-11-29)
+
+## Referenced Use Case(s)
+- [Common Configuration UCR](URL of use case link TBD)
+
+## Context
+This ADR describes the architecture of  the new common configuration capability which impacts all services. Requirements for this new capability are describe in the above referenced UCR. This is deemed architecturally significant due to the cross cutting impacts.
+
+### Current Design
+
+The following flow chart the bootstrapping of each services' configuration in the current Levski release.
+
+![](common-config-images/EdgeX 2.x Configuration bootstrapping flowchart.png)
+
+## Proposed Design
+The configuration settings that are common to all services will be partitioned out into a separate common configuration source. This  common configuration source will be pushed into the Configuration Provider by a new service. When the Configuration Provider is not used this new service will provide the common configuration via a RESTful endpoint.
+
+During bootstrapping, each service will either load the common configuration from the Configuration Provider or via URI to some endpoint that provides the common configuration. This could be the new service's REST endpoint. Each service will have additional private configuration, which may overwrite and/or extend the common configuration.  
+
+An additional common configuration setting must be present to indicate all other common settings have been pushed to the Configuration Provider. This setting is stored last and the services must wait for this setting to be present prior to pulling the common settings.
+
+### Configuration Bootstrapping
+
+The following flow chart demonstrates the bootstrapping of each services' configuration with this new common configuration capability.
+
+![](common-config-images/EdgeX 3.x Configuration bootstrapping flowchart.png)

```

Comment:
I think separating "common from CP" from "common from URI" adds unnecessary architecture complexity in favor of a simple solution which can't be extended for better security (e.g. adding access control, integrity checks to prevent man in the middle attacks). I was under the impression that we already agreed on doing that during the previous architecture call.